test/e2e/apimachinery/resource_quota.go

```diff
@@ -1186,21 +1186,36 @@ var _ = SIGDescribe("ResourceQuota", func() {
 		})
 		framework.ExpectNoError(err, "failed to locate ResourceQuota %q in namespace %q", patchedResourceQuota.Name, ns)
 
-		err = wait.PollUntilContextTimeout(ctx, 5*time.Second, 5*time.Minute, true, func(ctx context.Context) (bool, error) {
+		// the resource_quota_controller ignores changes to the status so we have to wait for a full resync of the controller
+		// to reconcile the status again, this full resync is set every 5 minutes by default so we need to poll at least one
+		// minute more just in case we we start to poll just after the full resync has happened and he have to wait until
+		// next full resync.
+		// Ref: https://issues.k8s.io/121911
+		err = wait.PollUntilContextTimeout(ctx, 5*time.Second, 6*time.Minute, true, func(ctx context.Context) (bool, error) {
 			resourceQuotaResult, err := rqClient.Get(ctx, rqName, metav1.GetOptions{})
-			framework.ExpectNoError(err)
+			if err != nil {
+				return false, nil
+			}
 
-			if apiequality.Semantic.DeepEqual(resourceQuotaResult.Spec.Hard.Cpu(), resourceQuotaResult.Status.Hard.Cpu()) {
-				gomega.Expect(*resourceQuotaResult.Status.Hard.Cpu()).To(gomega.Equal(resource.MustParse("1")), "Hard cpu value for ResourceQuota %q is %s not 1.", repatchedResourceQuota.Name, repatchedResourceQuota.Status.Hard.Cpu().String())
-				gomega.Expect(*resourceQuotaResult.Status.Hard.Memory()).To(gomega.Equal(resource.MustParse("1Gi")), "Hard memory value for ResourceQuota %q is %s not 1Gi.", repatchedResourceQuota.Name, repatchedResourceQuota.Status.Hard.Memory().String())
+			if *resourceQuotaResult.Spec.Hard.Cpu() == *resourceQuotaResult.Status.Hard.Cpu() {
```

Comment:
dereferencing and comparing rather than using Quantity#Equal seems fragile

Reply:
that is something I was wondering about, when I read the library I saw the euqality function but then in the same test I saw it was comparing the values, just for learning, what is the difference?

Reply:
my earlier comment:
https://github.com/kubernetes/kubernetes/pull/121951#discussion_r1398261216

the comparison by value is fine for now, but if the struct introduces pointer objects it would fail.

```go
package main

import "fmt"

type bar struct { i int }
type foo struct { b *bar }

func main() {
	a := &foo{b: &bar{i: 10}}
	b := &foo{b: &bar{i: 10}}
	fmt.Printf("a == b = %v", *a == *b) // a == b = false
}
```

Reply:
.Equal compares equality of the values, which may or may not be identical to struct equality, depending on normalization and whether the string rendering is cached

```go
func TestCPUComparison(t *testing.T) {
	l1 := ResourceList{"cpu": resource.MustParse("1m")}
	l2 := ResourceList{"cpu": resource.MustParse(".001")}

	cpu1 := l1.Cpu()
	cpu2 := l2.Cpu()

	t.Logf("%#v", cpu1)
	t.Logf("%#v", cpu2)

	if cpu1.Equal(*cpu2) {
		t.Log("passed Equal()")
	} else {
		t.Fatal("failed Equal()")
	}

	if *cpu1 == *cpu2 {
		t.Log("passed ==")
	} else {
		t.Fatal("failed ==")
	}
}
```

outputs

```sh
=== RUN   TestCPUComparison
    resource_test.go:16: &resource.Quantity{i:resource.int64Amount{value:1, scale:-3}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:"1m", Format:"DecimalSI"}
    resource_test.go:17: &resource.Quantity{i:resource.int64Amount{value:1, scale:-3}, d:resource.infDecAmount{Dec:(*inf.Dec)(nil)}, s:"", Format:"DecimalSI"}
    resource_test.go:20: passed Equal()
    resource_test.go:28: failed ==
--- FAIL: TestCPUComparison (0.00s)
```